spring-data-jdbc/src/main/java/org/springframework/data/jdbc/repository/config/AbstractJdbcConfiguration.java

```diff
@@ -147,23 +148,17 @@ public JdbcConverter jdbcConverter(JdbcMappingContext mappingContext, NamedParam
 	 * @return will never be {@literal null}.
 	 */
 	@Bean
-	public JdbcCustomConversions jdbcCustomConversions() {
-
-		try {
-
-			Dialect dialect = applicationContext.getBean(Dialect.class);
-			SimpleTypeHolder simpleTypeHolder = dialect.simpleTypes().isEmpty() ? JdbcSimpleTypes.HOLDER
-					: new SimpleTypeHolder(dialect.simpleTypes(), JdbcSimpleTypes.HOLDER);
-
-			return new JdbcCustomConversions(
-					CustomConversions.StoreConversions.of(simpleTypeHolder, storeConverters(dialect)), userConverters());
-
-		} catch (NoSuchBeanDefinitionException exception) {
-
+	public JdbcCustomConversions jdbcCustomConversions(Optional<Dialect> dialect) {
```

Comment:
This change does not affect the feature and introduces the non-compatible change. Let's segregate it into separate PR so it can be merged in the next major release

Reply:
OK, I'll revert this change for now, and then create another PR that adds a factory class later.